src/main.c

```diff
@@ -238,15 +238,15 @@ do_request(const void *client_config, const char *hostname, const char *path)
     goto cleanup;
   }
 
-  void *client_session = NULL;
-  int result =
+  rustls_client_session *client_session = NULL;
+  rustls_result result =
     rustls_client_session_new(client_config, hostname, &client_session);
-  if(result != CRUSTLS_OK) {
+  if(result != rustls_result_OK) {
```

Comment:
nit: I don't love the mixed case here (`rustls_result_ERROR`) but I suspect this enum has to be ugly on either the C side or the Rust side, so you probably can't do much. _Unless_ there's some kind of `#[repr]` tag you can put on the Rust enum so that the enum that gets rendered into the generated C header has SCREAMING_SNAKE_CASE? Like [`serde`'s `rename_all`](https://serde.rs/container-attrs.html)?

Reply:
Good call. This can be done with cbindgen.toml. Added.

Reply:
That's great, nice find!